From be7c1cc5906cd2804af29b90a23d97043a1f07f1 Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@nxp.com>
Date: Tue, 15 May 2018 15:28:01 +0800
Subject: [PATCH 3812/5242] MLK-18323-2 ARM64: dts: imx8qm: add ADC support

commit  642ba9bc47f99aebe168835d096e1e5a0a4dfdc4 from
https://source.codeaurora.org/external/imx/linux-imx.git

i.MX8QM contains two adc, and for imx8qm-mek board, only reserve one
pin for adc0 (ADC_IN0).

Signed-off-by: Haibo Chen <haibo.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-device.dtsi      |   28 ++++++++++++++++++++
 .../boot/dts/freescale/fsl-imx8qm-mek-domu.dts     |    2 ++
 arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts   |   20 ++++++++++++++
 3 files changed, 50 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
index a42a819..99709d8 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
@@ -2544,6 +2544,34 @@
 		};
 	};
 
+	adc0: adc@5a880000 {
+		compatible = "fsl,imx8qxp-adc";
+		reg = <0x0 0x5a880000 0x0 0x10000>;
+		interrupts = <GIC_SPI 240 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-parent = <&gic>;
+		clocks = <&clk IMX8QM_ADC0_CLK>,
+			 <&clk IMX8QM_ADC0_IPG_CLK>;
+		clock-names = "per", "ipg";
+		assigned-clocks = <&clk IMX8QM_ADC0_CLK>;
+		assigned-clock-rates = <24000000>;
+		power-domains = <&pd_dma_adc0>;
+		status = "disabled";
+	};
+
+	adc1: adc@5a890000 {
+		compatible = "fsl,imx8qxp-adc";
+		reg = <0x0 0x5a890000 0x0 0x10000>;
+		interrupts = <GIC_SPI 241 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-parent = <&gic>;
+		clocks = <&clk IMX8QM_ADC1_CLK>,
+			 <&clk IMX8QM_ADC1_IPG_CLK>;
+		clock-names = "per", "ipg";
+		assigned-clocks = <&clk IMX8QM_ADC1_CLK>;
+		assigned-clock-rates = <24000000>;
+		power-domains = <&pd_dma_adc1>;
+		status = "disabled";
+	};
+
 	i2c0: i2c@5a800000 {
 		compatible = "fsl,imx8qm-lpi2c", "fsl,imx7ulp-lpi2c";
 		reg = <0x0 0x5a800000 0x0 0x4000>;
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts
index 5877602..e77064d 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts
@@ -258,6 +258,8 @@
 /delete-node/ &ldb1;
 /delete-node/ &lvds1_pwm;
 /delete-node/ &camera;
+/delete-node/ &adc0;
+/delete-node/ &adc1;
 /delete-node/ &i2c0;
 /delete-node/ &i2c1;
 /delete-node/ &i2c2;
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
index 0b10844..b206958 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
@@ -74,6 +74,13 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 		};
+
+		reg_vref_1v8: adc_vref_1v8 {
+			compatible = "regulator-fixed";
+			regulator-name = "vref_1v8";
+			regulator-min-microvolt = <1800000>;
+			regulator-max-microvolt = <1800000>;
+		};
 	};
 
 	sound: sound {
@@ -209,6 +216,12 @@
 			>;
 		};
 
+		pinctrl_adc0: adc0grp {
+			fsl,pins = <
+				SC_P_ADC_IN0_DMA_ADC0_IN0		0xc0000060
+			>;
+		};
+
 		pinctrl_cm41_i2c0: cm41i2c0grp {
 			fsl,pins = <
 				SC_P_M41_I2C0_SCL_M41_I2C0_SCL          0x0600004c
@@ -544,6 +557,13 @@
 	};
 };
 
+&adc0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_adc0>;
+	vref-supply = <&reg_vref_1v8>;
+	status = "okay";
+};
+
 &asrc0 {
 	fsl,asrc-rate  = <48000>;
 	status = "okay";
-- 
1.7.9.5

